haystack/backends/solr_backend.py

```diff
@@ -389,13 +393,38 @@ def _process_results(self, raw_results, highlight=False, result_class=None, dist
                                                         facets[key][facet_field][1::2]))
 
         if self.include_spelling and hasattr(raw_results, 'spellcheck'):
-            # Solr 5+ changed the JSON response format so the suggestions will be key-value mapped rather
-            # than simply paired elements in a list, which is a nice improvement but incompatible with
-            # Solr 4: https://issues.apache.org/jira/browse/SOLR-3029
-            if len(raw_results.spellcheck.get('collations', [])):
-                spelling_suggestion = raw_results.spellcheck['collations'][-1]
-            elif len(raw_results.spellcheck.get('suggestions', [])):
-                spelling_suggestion = raw_results.spellcheck['suggestions'][-1]
+            # There are many different formats for Legacy, 6.4, and 6.5
+            # e.g. https://issues.apache.org/jira/browse/SOLR-3029
+            collations = raw_results.spellcheck.get('collations', [])
+            suggestions = raw_results.spellcheck.get('suggestions', [])
+            if len(collations):
+                #Handle sol6.5 collation format
```

Comment:
Minor nitpick but we should probably have a space after `#` in the comments in this block